<!-- -->
<launch>
  <arg name="engine_file" default="$(find lidar_cnn_seg_detect_trt)/../../../models/cnnseg/deploy.caffemodel_FP32.trt"/>
  <arg name="points_src" default="/middle/rslidar_points" />
  <arg name="publish_objects" default="/detection/lidar_detector/objects" />

  <arg name="objectness_thresh" default="0.7" />
  <arg name="height_thresh" default="0.5" />
  <arg name="min_pts_num" default="3" />
  <arg name="use_all_grids_for_clustering" default="false" />

  <arg name="score_threshold" default="0.75" /> 
  <arg name="min_height" default="-5.0" />
  <arg name="max_height" default="5.0" />
  <arg name="range" default="60.0" />
  <arg name="intensity_normalized" default="false" />
  <!-- <arg name="ground_height" default="-1.73" /> -->
  
  <arg name="benchmark_path" default="" />
  <arg name="duration_path" default="" />    

  <!-- for SHM  -->  
  <arg name="shm_enable" default="false" />  
  <arg name="shm_key" default="200" />   
  <arg name="sem_proj_id" default="100" />  
  <arg name="shm_clent_index" default="1" />   
  <arg name="shm_cycle_delay" default="30" />  

  <node pkg="lidar_cnn_seg_detect_trt" 
      type="lidar_cnn_seg_detect_trt" 
      name="lidar_cnn_seg_detect_trt_x64" 
      output="screen">
    <param name="engine_file" value="$(arg engine_file)" />
    <param name="points_src" value="$(arg points_src)" />    
    <param name="publish_objects" value="$(arg publish_objects)" />

    <param name="objectness_thresh" value="$(arg objectness_thresh)" />
    <param name="height_thresh" value="$(arg height_thresh)" />
    <param name="min_pts_num" value="$(arg min_pts_num)" />
    <param name="use_all_grids_for_clustering" value="$(arg use_all_grids_for_clustering)" />

    <param name="score_threshold" value="$(arg score_threshold)" />
    <param name="min_height" value="$(arg min_height)" />
    <param name="max_height" value="$(arg max_height)" />
    <param name="range" value="$(arg range)" />
    <param name="intensity_normalized" value="$(arg intensity_normalized)" />
    <!-- <param name="ground_height" value="$(arg ground_height)" />     -->

    <param name="benchmark_path" value="$(arg benchmark_path)" />
    <param name="duration_path" value="$(arg duration_path)" />

    <!-- for SHM  -->  
    <param name="shm_enable" value="$(arg shm_enable)" />
    <param name="shm_key" value="$(arg shm_key)" />
    <param name="sem_proj_id" value="$(arg sem_proj_id)" />
    <param name="shm_clent_index" value="$(arg shm_clent_index)" />
    <param name="shm_cycle_delay" value="$(arg shm_cycle_delay)" />

  </node>

  <node pkg="detected_objects_visualizer" 
      type="visualize_detected_objects" 
      name="cluster_detect_visualization_02"
      output="screen" ns="/detection/lidar_detector" />

</launch>
